pkgs/development/tools/build-managers/bazel/nix-hacks.patch

```diff
@@ -1,31 +1,3 @@
-diff -Naur a/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java b/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java
---- a/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java	2019-06-12 20:39:37.420705161 -0700
-+++ b/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java	2019-06-12 20:44:18.894429744 -0700
-@@ -428,24 +428,7 @@
-       try {
-         content = FileSystemUtils.readContent(markerPath, StandardCharsets.UTF_8);
-         String markerRuleKey = readMarkerFile(content, markerData);
--        boolean verified = false;
--        if (Preconditions.checkNotNull(ruleKey).equals(markerRuleKey)
--            && Objects.equals(
--                markerData.get(MANAGED_DIRECTORIES_MARKER),
--                this.markerData.get(MANAGED_DIRECTORIES_MARKER))) {
--          verified = handler.verifyMarkerData(rule, markerData, env);
--          if (env.valuesMissing()) {
--            return null;
--          }
--        }
--
--        if (verified) {
-           return new Fingerprint().addString(content).digestAndReset();
--        } else {
--          // So that we are in a consistent state if something happens while fetching the repository
--          markerPath.delete();
--          return null;
--        }
-       } catch (IOException e) {
-         throw new RepositoryFunctionException(e, Transience.TRANSIENT);
-       }
```

Comment:
Why is ignoring the markers not necessary anymore? I thought it was necessary because we patch the markers to make them deterministic, but was it only necessary in the past maybe because we deleted the markers instead of patching them? I'm not comfortable with removing code without understand why it's there in the first place, can you find out?

Reply:
Indeed, this works now because we just delete the markers instead of patching them. I suppose when I implemented this back in fe153d73ce3add3f0733469c40b1748cd0743e5f and ba9cde1dd55532f673c3fd64bce62290c19b4e32 I either didn't try just removing the markers (?) or somehow it didn't work.

Reply:
> 2. Build an actual derivation which uses fetched dependencies (skipping checksums needed here because they depend on the build directory).
This might also be a hint, maybe Bazel changed their checksum algorithm since then to not depend on something? Maybe inodes?

Reply:
I expect deleting the markers to not work, but patching them correctly should. Sounds good to me if you've tested it and it doesn't cause bazel to download the dependencies again after the initial fetch.

Reply:
BTW see also https://github.com/NixOS/nixpkgs/pull/64904 , I splitted it to a separate PR.